google/cloud/spanner/internal/session_pool.cc

```diff
@@ -141,6 +141,14 @@ StatusOr<SessionHolder> SessionPool::Allocate(bool dissociate_from_pool) {
   }
 }
 
+void SessionPool::AssignStubIfNeeded(SessionHolder& session) {
+  if (session->stub() == nullptr) {
```

Comment:
Does it make any additional sense to move this logic into `Session`.  That is, `session->AssignStubIfNeeded(stub_)`?